### PR TITLE
Use explicit path to rusts executables

### PR DESCRIPTION
**Pull request by b52**
This way we don't make any assumption on the installation of rust.
Additionally, the default values provide a flawless setup on properly
configured Windows, Linux and OSX boxes.

### CODE REVIEW

File: lib/init.coffee [hunk 1]
```diff
@@ -2,10 +2,14 @@
 
 module.exports =
   config:
-    rustHome:
+    rustcPath:
       type: 'string'
-      default: '/usr/local'
-      description: 'Path to Rust\'s home directory. rustc should exist in /bin/rustc from here.'
+      default: 'rustc'
+      description: 'Path to Rust\'s compiler `rustc`.'
```
Comment (steelbrain) [inline lib/init.coffee]: How about you replace `'` with `"` to avoid escaping here

File: lib/linter-rust.coffee [hunk 2]
```diff
@@ -1,7 +1,8 @@
 fs = require 'fs'
 path = require 'path'
 {BufferedProcess} = require 'atom'
-XRegExp = require('xregexp').XRegExp
+{spawn} = require 'child_process'
```
Comment (steelbrain) [inline lib/linter-rust.coffee]: looks like a redundant import

File: lib/init.coffee [hunk 3]
```diff
@@ -35,9 +39,13 @@ module.exports =
       Please install https://github.com/zargony/atom-language-rust'
 
     @subscriptions = new CompositeDisposable
-    @subscriptions.add atom.config.observe 'linter-rust.rustHome',
-    (rustHome) =>
-      @rustHome = rustHome
+    @subscriptions.add atom.config.observe 'linter-rust.rustcPath',
+    (rustcPath) =>
```
Comment (steelbrain) [inline lib/init.coffee]: can this be on the same line please?

**Comment by b52** [comment]
Fixed your remarks @steelbrain

**Comment by steelbrain** [comment]
LGTM. Thanks for your contributions :clap:
/cc @gmist

**Comment by steelbrain** [comment]
@b52 Your PR has some merge conflicts that you might wanna squash. Bonus: Atom has a [package][1] for that.
[1]:atom.io/packages/merge-conflicts

**Comment by b52** [comment]
@steelbrain Solved.

**Comment by steelbrain** [comment]
LGTM
/cc @gmist

**Comment by b52** [comment]
Fixed merge conflicts once again.
@gmist Do you mind merging this PR or explaining why you don't want to? A little bit more collaboration would be nice.

**Comment by steelbrain** [comment]
Enjoy :wink:

**Comment by gmist** [comment]
@steelbrain I had some reason not to merge this PR... but let be so.

**Comment by steelbrain** [comment]
sorry I thought you were busy, 'cause we hadn't heard from you for quite some time :)

**Comment by b52** [comment]
@gmist Do you mind sharing the reason?

**Comment by gmist** [comment]
The idea is to find the path to executable files (rustc/cargo) automatically if they are in env.PATH and not to force the user to enter the paths to the rustc/cargo separately if he wants to use a specific version rustc/cargo (that is not added to the env.PATH) - this would allow to quickly switch between multiple Rust installations (stable/nightly).

**Comment by b52** [comment]
The old solution forced me to enter the path to my rust 'home', because the default was '/usr/local' and I was running a Windows setup. Using my setup, there is no initial setup required on any platform, assuming you have a sanely configured `PATH`, because the executables are configurable and default to the executable names, i.e., `cargo` and `rustc`.

However, I agree that you are now forced to edit one more path, if you happen to switch between versions very often. Although, I assume the required workload is negligible, i.e., two character changes vs four character changes, maybe. This fact and the more pleasant initial setup let me believe the change is worth it.

**Comment by gmist** [comment]
Probably you're right and required workload is negligible because everyone has copy/paste.

**Pull request merged**